http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriberSpec.scala

```diff
@@ -1781,6 +1781,60 @@ class LeoPubsubMessageSubscriberSpec
     res.unsafeRunSync()(cats.effect.unsafe.IORuntime.global)
   }
 
+  it should "handle Azure StartRuntimeMessage and start runtime" in isolatedDbTest {
+    val azurePubSubHandler = makeAzureInterp()
+    val leoSubscriber = makeLeoSubscriber(azureInterp = azurePubSubHandler)
+
+    val res = for {
+      disk <- makePersistentDisk().copy(status = DiskStatus.Ready).save()
+
+      azureRuntimeConfig = RuntimeConfig.AzureConfig(MachineTypeName(VirtualMachineSizeTypes.STANDARD_A1.toString),
+                                                     disk.id,
+                                                     azureRegion
+      )
+
+      runtime <- IO(
+        makeCluster(1)
+          .copy(status = RuntimeStatus.Starting, cloudContext = CloudContext.Azure(azureCloudContext))
+          .saveWithRuntimeConfig(azureRuntimeConfig)
+      )
+      tr <- traceId.ask[TraceId]
+      _ <- leoSubscriber.messageResponder(StartRuntimeMessage(runtime.id, Some(tr)))
+      updatedRuntime <- clusterQuery.getClusterById(runtime.id).transaction
+    } yield {
+      updatedRuntime shouldBe defined
+      updatedRuntime.get.status shouldBe RuntimeStatus.Starting
+    }
+    res.unsafeRunSync()(cats.effect.unsafe.IORuntime.global)
+  }
+
+  it should "handle Azure StopRuntimeMessage and stop runtime" in isolatedDbTest {
+    val azurePubSubHandler = makeAzureInterp()
+    val leoSubscriber = makeLeoSubscriber(azureInterp = azurePubSubHandler)
+
+    val res = for {
+      disk <- makePersistentDisk().copy(status = DiskStatus.Ready).save()
+
+      azureRuntimeConfig = RuntimeConfig.AzureConfig(MachineTypeName(VirtualMachineSizeTypes.STANDARD_A1.toString),
+                                                     disk.id,
+                                                     azureRegion
+      )
+
+      runtime <- IO(
+        makeCluster(1)
+          .copy(status = RuntimeStatus.Stopping, cloudContext = CloudContext.Azure(azureCloudContext))
+          .saveWithRuntimeConfig(azureRuntimeConfig)
+      )
+      tr <- traceId.ask[TraceId]
+      _ <- leoSubscriber.messageResponder(StopRuntimeMessage(runtime.id, Some(tr)))
+      updatedRuntime <- clusterQuery.getClusterById(runtime.id).transaction
+    } yield {
+      updatedRuntime shouldBe defined
```

Comment:
same comment

Reply:
I think most of the logic lives in `AzurePubsubHandler`, so I think the unit tests `AzurePubsubHandlerSpec` is mostly enough...unless you have some specific logic in `leoPubsubMessageSubscriber` that you'd like to test

Reply:
if you wanna test out branching logic, use mock of the AzurePubsubHandler and confirm the mock is being called will be much better test than just validating a DB record never changed (which to me is pretty useless)